[question] How to handle *.cmake macros in my package

### What is your question?

Hello,
My package besides libs and include files contains two cmake files. Let's say they are Macros1.cmake Macros2.cmake.
And ofc I would like to have access to them in my test_package. I can not find anything like this on the internet.
I beg for the help...

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

Okey, I have tried for an example something like this:
```py
        copy(self, "Macros.cmake", src=os.path.join(self.source_folder, "cmake_tools"), dst=os.path.join(self.package_folder, "cmake_tools"))
        ...
        cmake = CMake(self)
        cmake.install()

    def package_info(self):
        self.cpp_info.components["cmake_tools"].builddirs = ["cmake_tools"]
        self.cpp_info.components["cmake_tools"].set_property("cmake_target_name", "cmake_tools")
        
        self.cpp_info.components["mylib"].libs = ["mylib"]
        ...
```
```cmake
project(test_package)

find_package(conan-package-name REQUIRED COMPONENTS cmake_tools)
include(Macros.cmake)

mymacro()
   ```
and it throws 
```
cmake] CMake Error at CMakeLists.txt:7 (include):
[cmake]   include could not find requested file:
[cmake] 
[cmake]     Macros.cmake
```
Can you help?
@memsharded

--- a/conans/test/functional/graph/test_transitive_build_scripts.py
+++ b/conans/test/functional/graph/test_transitive_build_scripts.py
@@ -83,3 +83,64 @@ def test_transitive_build_scripts():
 
     assert "MYFUNCTION CMAKE B: Hello world B!!!" in c.out
     assert "MYFUNCTION CMAKE A: Hello world A!!!" in c.out
+
+
+@pytest.mark.tool("cmake")
+def test_reuse_macro_from_dep():
+    """
+    A simple cmake script can be added from a regular requires, without even a ``find_package()``
+    https://github.com/conan-io/conan/issues/14013
+    """
+    c = TestClient()
+    pkg = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.files import copy
+        class Pkg(ConanFile):
+            name = "pkg"
+            version = "0.1"
+            package_type = "static-library"
+            exports_sources = "*.cmake"
+            def package(self):
+                copy(self, "*.cmake", src=self.source_folder, dst=self.package_folder)
+            def package_info(self):
+                self.cpp_info.builddirs = ["."]
+            """)
+    pkg_macros = textwrap.dedent("""
+        function(pkg_macro)
+            message(STATUS "PKG MACRO WORKING!!!")
+        endfunction()
+    """)
+    app = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.cmake import CMake
+
+        class App(ConanFile):
+            package_type = "application"
+            generators = "CMakeToolchain"
+            settings = "os", "compiler", "arch", "build_type"
+
+            def requirements(self):
+                self.requires("pkg/0.1")
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+                cmake.build()
+        """)
+    app_cmake = textwrap.dedent("""
+        cmake_minimum_required(VERSION 3.15)
+        project(App LANGUAGES NONE)
+
+        include(Macros)
+        pkg_macro()
+    """)
+    c.save({"pkg/conanfile.py": pkg,
+            "pkg/Macros.cmake": pkg_macros,
+            "app/conanfile.py": app,
+            "app/CMakeLists.txt": app_cmake})
+
+    c.run("create pkg")
+    c.run("build app")
+
+    assert "PKG MACRO WORKING!!!" in c.out
+

